io/include/reader.hpp

```diff
@@ -14,22 +15,27 @@ using namespace std::placeholders;
 
 namespace traccc {
 
-const std::string data_directory() {
-    auto env_d_d = std::getenv("TRACCC_TEST_DATA_DIR");
-    if (env_d_d == nullptr) {
-        throw std::ios_base::failure(
-            "Test data directory not found. Please set TRACCC_TEST_DATA_DIR.");
-    }
-    std::string data_dir = std::string(env_d_d);
-    return data_dir.append("/");
+const std::string &data_directory() {
```

Comment:
I'm not sure I am too thrilled about returning a static reference here. It shouldn't be a big problem for now, though.

Reply:
Hm, what's your concern? Lifetime should be fine, and we don't have to copy a string that's not supposed to change over any executable lifetime.

Reply:
I guess if we only ever have one data directory per executable lifetime (which does make sense), then it's fine.